test/api/pages/AbstractPage.ts

```diff
@@ -0,0 +1,30 @@
+import { expect, Locator, Page } from '@playwright/test';
+
+export class AbstractPage {
+
+    readonly page: Page;
+    readonly url?: string;
+    readonly atChecker?: Locator;
+
+    constructor(pageInit: Page, urlInit?: string, atChecker?: Locator) {
+        this.page = pageInit;
+        this.url = urlInit;
+        this.atChecker = atChecker
+    }
+
+    async goto() {
+        if (this.url) {
+            await this.page.goto(this.url);
+        }
+    }
```

Comment:
in theory, we should not use this type of navigation in a SPA, so having it as function on every Prototype chain seems excessive, but not harmful

Reply:
Bookmarkable state can be important even in SPAs.
I do not see any problem with this, although it is true that I got this code from a test system of a different kind of application.